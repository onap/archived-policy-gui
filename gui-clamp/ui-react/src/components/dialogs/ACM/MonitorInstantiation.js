/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import InstantiationItem from "./InstantiationItem";
import ACMService from "../../../api/ACMService";
import InstantiationElements from "./InstantiationElements";
import { Alert } from "react-bootstrap";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`
const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const MonitorInstantiation = (props) => {
  const [show, setShow] = useState(true);
  const [acmList, setAcmList] = useState([]);
  const [acmInstantiationOk, setAcmInstantiationOk] = useState(true);
  const [acmInstantiationError, setACMInstantiationError] = useState({});

  useEffect(async () => {
    const acmInstantiation = await ACMService.getACMInstantiation()
      .catch(error => error.message);

    const acmInstantiationJson = await acmInstantiation.json();

    if (!acmInstantiation.ok || acmInstantiationJson['automationCompositionList'].length === 0) {
      setAcmInstantiationOk(false)
      setACMInstantiationError(acmInstantiationJson)
    } else {
      setAcmList(acmInstantiationJson['automationCompositionList']);
    }
  }, [])

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  return (
    <ModalStyled size="xl" show={ show } onHide={ handleClose } backdrop="static" keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Tosca Instantiation - Monitoring</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {
          acmList.map((clList, index) => (
            <InstantiationItem title={ clList["name"] } orderedState={ clList["orderedState"] } index={ index } key={ index } >
              <InstantiationElements elements={ clList["elements"] } />
            </InstantiationItem>
          ))
        }
        <AlertStyled show={ !acmInstantiationOk }
                     variant="danger">Can't get acm instantiation info:<br/>{ JSON.stringify(acmInstantiationError, null, 2) }</AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" type="null" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  )
}

export default MonitorInstantiation;
