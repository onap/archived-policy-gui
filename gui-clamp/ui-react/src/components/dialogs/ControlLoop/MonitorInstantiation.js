/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useEffect, useState } from "react";
import styled from "styled-components";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import InstantiationItem from "./InstantiationItem";
import ControlLoopService from "../../../api/ControlLoopService";
import InstantiationElements from "./InstantiationElements";
import { Alert } from "react-bootstrap";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`
const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const MonitorInstantiation = (props) => {
  const [show, setShow] = useState(true);
  const [controlLoopList, setControlLoopList] = useState([]);
  const [controlLoopInstantiationOk, setControlLoopInstantiationOk] = useState(true);
  const [controlLoopInstantiationError, setControlLoopInstantiationError] = useState({});

  useEffect(async () => {
    const controlLoopInstantiation = await ControlLoopService.getControlLoopInstantiation()
      .catch(error => error.message);

    const controlLoopInstantiationJson = await controlLoopInstantiation.json()
    console.log(controlLoopInstantiationJson)
    if (!controlLoopInstantiation.ok || controlLoopInstantiationJson['controlLoopList'].length === 0) {
      setControlLoopInstantiationOk(false)
      setControlLoopInstantiationError(controlLoopInstantiationJson)
    } else {
      setControlLoopList(controlLoopInstantiationJson[['controlLoopList']])
    }
  }, [])

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  return (
    <ModalStyled size="xl" show={ show } onHide={ handleClose } backdrop="static" keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Tosca Instantiation - Monitoring</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        {
          controlLoopList.map((clList, index) => (
            <InstantiationItem title={ clList["name"] } orderedState={ clList["orderedState"] } index={ index } key={ index }>
              <InstantiationElements elements={ clList["elements"] } />
            </InstantiationItem>
          ))
        }
        <AlertStyled show={ !controlLoopInstantiationOk }
                     variant="danger">Can't get control loop instantiation info:<br/>{ JSON.stringify(controlLoopInstantiationError, null, 2) }</AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" type="null" onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  )
}

export default MonitorInstantiation;
