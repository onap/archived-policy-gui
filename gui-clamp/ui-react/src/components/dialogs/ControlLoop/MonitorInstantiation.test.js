/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { createMemoryHistory } from "history";
import React from "react";
import MonitorInstantiation from "./MonitorInstantiation";
import { act } from "react-dom/test-utils";

describe('Verify MonitoringInstantiation', () => {

  it("renders without crashing", () => {
    shallow(<MonitorInstantiation/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<MonitorInstantiation/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have a Button element', () => {
    const container = shallow(<MonitorInstantiation/>)
    expect(container.find('Button').length).toEqual(1);
  });

  it('handleClose called when bottom button clicked', () => {
    const history = createMemoryHistory();
    const component = mount(<MonitorInstantiation history={ history }/>)
    const logSpy = jest.spyOn(console, 'log');

    act(() => {
      component.find('[variant="secondary"]').simulate('click');
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });

  it('handleClose called when top-right button clicked', () => {
    const history = createMemoryHistory();
    const component = mount(<MonitorInstantiation history={ history }/>)
    const logSpy = jest.spyOn(console, 'log');

    act(() => {
      component.find('[size="xl"]').get(0).props.onHide();
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });
});
