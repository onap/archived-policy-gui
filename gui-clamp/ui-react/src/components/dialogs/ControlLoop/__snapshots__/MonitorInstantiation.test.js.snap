// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify MonitoringInstantiation renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Tosca Instantiation - Monitoring
    </ModalTitle>
  </ModalHeader>
  <ModalBody />
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
